sympy/logic/boolalg.py

```diff
@@ -1915,6 +2093,213 @@ def POSform(variables, minterms, dontcares=None):
+    Examples
+    ========
+    >>> from sympy.logic.boolalg import ANFform
+    >>> from sympy.abc import x, y
+    >>> ANFform([x], [1, 0])
+    Xor(x, True)
+    >>> ANFform([x, y], [0, 1, 1, 1])
+    Xor(x, y, x & y)
+
+    References
+    ==========
+
+    .. [2] en.wikipedia.org/wiki/Zhegalkin_polynomial
+
+    """
+    variables = [sympify(v) for v in variables]
+
+    coeffs = anf_coeffs(truthvalues)
+    terms = []
+
+    for i, t in enumerate(product([0, 1], repeat=len(variables))):
+        if coeffs[i] == 1:
+            terms.append(t)
+
+    return Xor(*[_convert_to_varsANF(x, variables) for x in terms],\
+            remove_true=False)
+
+
+def anf_coeffs(truthvalues):
+    """
+    Convert a list of truth values of some boolean expression
+    to the list of coefficients of the polynomial mod 2 (exclusive
+    disjunction) representing the boolean expression in ANF
+    (i.e., the "Zhegalkin polynomial").
+
+    There are 2^n possible Zhegalkin monomials in n variables, since
+    each monomial is fully specified by the presence or absence of
+    each variable.
+
+    A given monomial's presence or absence in a polynomial corresponds
+    to that monomial's coefficient being 1 or 0 respectively.
+
+    Examples
+    ========
+    >>> from sympy.logic.boolalg import anf_coeffs, monomial, Xor
+    >>> from sympy.abc import a, b, c
+    >>> truthvalues = [0, 1, 1, 0, 0, 1, 0, 1]
+    >>> coeffs = anf_coeffs(truthvalues)
+    >>> coeffs
+    [0, 1, 1, 0, 0, 0, 1, 0]
+    >>> polynomial = Xor(*[monomial(k, [a, b, c])
+    ...         for k, coeff in enumerate(coeffs) if coeff==1])
+    >>> polynomial
+    Xor(b, c, a & b)
+
+    """
+
+    s = '{0:b}'.format(len(truthvalues))
+    n = len(s) - 1
+
+    if len(truthvalues) != 2**n:
+        raise ValueError('The number of truth values must be a power of two.')
+
+    coeffs = [[v] for v in truthvalues]
+
+    for i in range(n):
+        tmp = []
+        for j in range(2 ** (n-i-1)):
+            tmp.append(coeffs[2*j] +
+                list(map(lambda x, y: x^y, coeffs[2*j], coeffs[2*j+1])))
+        coeffs = tmp
+
+    return coeffs[0]
+
+
+def minterm(k, variables):
+    """
+    Return the k-th minterm.
+
+    Minterms are numbered by a binary encoding of the complementation
+    pattern of the variables. This convention assigns the value 1 to
+    the direct form and 0 to the complemented form.
+
+    Parameters
+    ==========
+
+    k : int or list of 1s and 0s (complementation patter)
+    variables : list of variables
+
+    Examples
+    ========
+    >>> from sympy.logic.boolalg import minterm
+    >>> from sympy.abc import x, y, z
+    >>> minterm([1, 0, 1], [x, y, z])
+    x & z & ~y
+    >>> minterm(6, [x, y, z])
+    x & y & ~z
+
+    References
+    ==========
+
+    .. [3] en.wikipedia.org/wiki/Canonical_normal_form#Indexing_minterms
+
+    """
+    if isinstance(k, int):
+        k = integer_to_term(k, len(variables))
+    variables = [sympify(v) for v in variables]
```

Comment:
use `map` here.

Reply:
If I use map I get: "TypeError: 'map' object is not subscriptable"

Reply:
I mean `list(map(sympify, variables))`

Reply:
Done.